pkg/proxy/iptables/proxier.go

```diff
@@ -1472,25 +1499,64 @@ func (proxier *Proxier) syncProxyRules() {
 				}
 			}
 
-			// Setup probability filter rules only over local endpoints
-			for i, endpointChain := range localEndpointChains {
+			for i, endpointChain := range localReadyEndpointChains {
 				// Balancing rules in the per-service chain.
 				args = append(args[:0],
 					"-A", string(svcXlbChain),
 					"-m", "comment", "--comment",
 					fmt.Sprintf(`"Balancing rule %d for %s"`, i, svcNameString),
 				)
-				if i < (numLocalEndpoints - 1) {
+				if i < (numLocalReadyEndpoints - 1) {
 					// Each rule is a probabilistic match.
 					args = append(args,
 						"-m", "statistic",
 						"--mode", "random",
-						"--probability", proxier.probability(numLocalEndpoints-i))
+						"--probability", proxier.probability(numLocalReadyEndpoints-i))
 				}
 				// The final (or only if n == 1) rule is a guaranteed match.
 				args = append(args, "-j", string(endpointChain))
 				writeLine(proxier.natRules, args...)
 			}
+		} else if numLocalServingTerminatingEndpoints > 0 {
```

Comment:
I can't tell if this block is identical to the above block or not - we need to de-dupe them and make this a single block of logic, I think,

Reply:
They are almost identical, the difference is that one iterate over `localReadyEndpointChains` while the other iterates `localServingTerminatingEndpointChains`

Reply:
one other difference is that probability is based on numLocalReadyEndpoints vs numLocalServingTerminatingEndpoints

Reply:
Can we just parameterize and use the same code block rather than duplicating?

Reply:
Yeah I can do that